src/OpenFeatureSDK/Model/Structure.cs

```diff
@@ -1,32 +1,45 @@
-using System;
 using System.Collections;
 using System.Collections.Generic;
+using System.Collections.Immutable;
 using System.Diagnostics.CodeAnalysis;
 
 namespace OpenFeatureSDK.Model
 {
     /// <summary>
     /// Structure represents a map of Values
     /// </summary>
-    public class Structure : IEnumerable<KeyValuePair<string, Value>>
+    public sealed class Structure : IEnumerable<KeyValuePair<string, Value>>
     {
-        private readonly Dictionary<string, Value> _attributes;
+        private readonly ImmutableDictionary<string, Value> _attributes;
 
         /// <summary>
-        /// Creates a new structure with an empty set of attributes
+        /// Internal constructor for use by the builder.
         /// </summary>
-        public Structure()
+        internal Structure(ImmutableDictionary<string, Value> attributes)
         {
-            this._attributes = new Dictionary<string, Value>();
+            this._attributes = attributes;
         }
 
+        /// <summary>
+        /// Private constructor for creating an empty <see cref="Structure"/>.
+        /// </summary>
+        private Structure()
+        {
+            this._attributes = ImmutableDictionary<string, Value>.Empty; ;

```

Comment:
Remove an extra semi-colon
```suggestion
            this._attributes = ImmutableDictionary<string, Value>.Empty;

```

Reply:
Apparently this line of code was sad about something. Committing your suggestion.